internal/peer/channel/joinbysnapshotstatus_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package channel
+
+import (
+	"testing"
+	"time"
+
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/internal/peer/common"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+)
+
+func TestJoinBySnapshotStatus(t *testing.T) {
+	defer viper.Reset()
+	defer resetFlags()
+
+	InitMSP()
+	signer, err := common.GetDefaultSigner()
+	require.NoError(t, err)
+
+	mockResponse := &pb.ProposalResponse{
+		Response:    &pb.Response{Status: 200},
+		Endorsement: &pb.Endorsement{},
+	}
+	mockEndorserClient := common.GetMockEndorserClient(mockResponse, nil)
+	mockCF := &ChannelCmdFactory{
+		EndorserClient:   mockEndorserClient,
+		BroadcastFactory: mockBroadcastClientFactory,
+		Signer:           signer,
+	}
+
+	// successful joinbysnapshotstatuscmd test
+	resetFlags()
+	cmd := joinBySnapshotStatusCmd(mockCF)
+	AddFlags(cmd)
+	require.NoError(t, cmd.Execute())
+
+	// very joinbysnapshotstatus returns correct value
+	client := &endorserClient{mockCF}
+	mockResponse.Response.Payload = []byte("")
+	status, err := client.joinBySnapshotStatus()
+	require.NoError(t, err)
+	require.Equal(t, "", status)
+
+	mockResponse.Response.Payload = []byte("mock_snapshot_directory")
+	status, err = client.joinBySnapshotStatus()
+	require.NoError(t, err)
+	require.Equal(t, "mock_snapshot_directory", status)
+
```

Comment:
It would have been better to cover the full output of the command by passing the writer from outside instead of using fmt.Println in the command code. But, it seems that all the commands in this package follow this pattern. To take a note of this, I think that it would be better if in a future PR, the command factory can be enhanced to pass the io.writer for this purpose.

Reply:
Agree. Will open a jira for this improvement.